### ISSUE TITLE
[ISSUE] Issue with `databricks_job` resource

### ISSUE THREAD

**Issue by ellajohnsen** [databricks/terraform-provider-databricks#3126]
### Configuration

```hcl
resource "databricks_job" "this" {
  name = "Test_Job"


  task {
    task_key        = "ZZ_test"
    job_cluster_key = "my-cluster"

    notebook_task {
      notebook_path = "test_task_key_order/helloworld"
    }
  }

  task {
    task_key        = "AA_test"
    job_cluster_key = "my-cluster"

    depends_on {
      task_key = "ZZ_test"
    }

    notebook_task {
      notebook_path = "test_task_key_order/helloworld"
    }
  }

  job_cluster {
    job_cluster_key = "my-cluster"
    new_cluster {
      spark_version       = "13.3.x-cpu-ml-scala2.12"
      node_type_id        = "Standard_F4s"
      driver_node_type_id = "Standard_F4s"
      spark_env_vars      = local.env_vars
      runtime_engine      = "STANDARD"

      custom_tags = local.job_tags

      autoscale {
        min_workers = 1
        max_workers = 20
      }

      spark_conf = local.spark_conf
    }
  }

}

```

### Expected Behavior

When creating a new change, unrelated to this resource, such as adding a new variable, the terraform plan for this resource should show no changes.

This is because I thought that PR https://github.com/databricks/terraform-provider-databricks/pull/3000 would ensure that the `task_key`s would be sorted by `depends_on` and not alphabetically.

### Actual Behavior

```
Terraform will perform the following actions:

  # module.test_task_key_order.databricks_job.this will be updated in-place
  ~ resource "databricks_job" "this" {
        id                        = "<redacted>"
        name                      = "Test_Job"
        tags                      = {<redacted> }
        # (9 unchanged attributes hidden)

      ~ task {
          ~ task_key                  = "AA_test" -> "ZZ_test"
            # (6 unchanged attributes hidden)

          - depends_on {
              - task_key = "ZZ_test" -> null
            }

            # (2 unchanged blocks hidden)
        }
      ~ task {
          ~ task_key                  = "ZZ_test" -> "AA_test"
            # (6 unchanged attributes hidden)

          + depends_on {
              + task_key = "ZZ_test"
            }

            # (2 unchanged blocks hidden)
        }

        # (7 unchanged blocks hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

### Steps to Reproduce

`terraform apply`

### Terraform and provider versions

```
Terraform v1.5.7
on darwin_arm64
+ provider registry.terraform.io/databricks/databricks v1.34.0
+ provider registry.terraform.io/hashicorp/azurerm v3.87.0
```

### Is it a regression?

No, this hasn't worked as far as I know. But I thought that the newest version of the `databricks provider`, 1.34.0, would solve this issue.

**Comment by mgyucht** [comment]
I believe #3000 only applied to the sorting of `depends_on` blocks within a task, not the tasks themselves.

I think the provider should be able to sort the tasks in the response from the Jobs API in the same order as specified by the user to prevent this from happening. Does that make sense @alexott?

**Comment by alexott** [comment]
Tasks should be sorted already using the topological sort based on `depends_on`

**Comment by ellajohnsen** [comment]
@alexott @mgyucht 
I also understood this as a topological sort based on a depends on yes, but that does not seem to be working. I have described to tasks `ZZ_task` and `AA_task`, where `AA_task` depends on `ZZ_task`, so I would expect the order to be
1. ZZ_task
2. AA_task

However, it seems like the alphabetical sort of the task keys are overwriting this. I thought that PR #3000 was supposed to fix this issue but maybe I'm wrong?

**Comment by case-k-git** [comment]
>I believe https://github.com/databricks/terraform-provider-databricks/pull/3000 only applied to the sorting of depends_on blocks within a task, not the tasks themselves.

Yea not the tasks themselves. fixing the order within the depends_on blocks inside the task. So example will be like this.

```
  task {
      task_key = "f"

      depends_on {
        task_key = "a"
      }
      depends_on {
        task_key = "b"
      }
      depends_on {
        task_key = "c"
      }
      depends_on {
        task_key = "d"
      }
      depends_on {
        task_key = "e"
      }
  }
```

**Comment by case-k-git** [comment]
Yea this issue is not  the one which solved by #300 but the issue is exist. So may be you do not need to close the issue
